server-spi-private/src/main/java/org/keycloak/policy/NotContainsUsernamePasswordPolicyProvider.java

```diff
@@ -36,7 +36,7 @@ public PolicyError validate(String username, String password) {
         if (username == null) {
```

Comment:
Could in any cases `password` be null? If so, a similar null check would be a good idea to prevent unexpected `NullPointerException`s from being raised when `password.contains(username)` is called.

Reply:
In fact, I think even this check is unnecessary once no other provider does that. Others providers uses username and password without any check.  I think is better just remove that.

@pedroigor wdty?

Reply:
Actually other providers, such as `NotUsernamePasswordPolicyProvider` checks for the `null` username, but none check for a `null` password. In this case, it should be irrelevant if username is `null` or not because `equals` will be `false` if it is, so the check can be skipped completely, IMO.

Reply:
@csakyzsolt It makes sense to me. We can do it as a follow-up. If you want to send a contribution to fix this, I'm happy to review/approve it.